File: SampleApps/ReceiverDemo/DPLProductRouteHandler.m
#import "DPLProductRouteHandler.h"
#import "DPLProductDetailViewController.h"

@implementation DPLProductRouteHandler

- (UIViewController <DPLTargetViewController> *)targetViewController {
    UIStoryboard *storyboard = [UIApplication sharedApplication].keyWindow.rootViewController.storyboard;
    
    return [storyboard instantiateViewControllerWithIdentifier:@"detail"];
}

@end

File: SampleApps/ReceiverDemoSwift/RouteHandlers/DPLProductRouteHandler.swift
import Foundation

open class DPLProductRouteHandler: DPLRouteHandler {
    open override func targetViewController() -> UIViewController & DPLTargetViewController {
        if let storyboard = UIApplication.shared.keyWindow?.rootViewController?.storyboard {
            return storyboard.instantiateViewController(withIdentifier: "detail") as! DPLProductDetailViewController
        }
        
        return super.targetViewController()!
    }
}
